scripts/ci/docker_upgrade.sh

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env bash
+set -e
+
+check_parameters() {
+  echo "-- check_parameters"
+
+  if [ -z "${IMAGE_NAME}" ] || [ -z "${IMAGE_VARIANT}" ] || [ -z "${DOCKER_REPOSITORY}" ]; then
+    echo "\${IMAGE_NAME}, \${IMAGE_VARIANT} and \${DOCKER_REPOSITORY} need to be defined."
+
+    false
+  fi
+}
+
+check_version() {
+  echo "-- check_version"
+
+  git fetch
+  git branch
+  DOCKERFILE="scripts/docker/${IMAGE_NAME}/Dockerfile.${IMAGE_VARIANT}"
+  PREV_VERSION=$(git diff origin/develop HEAD -- "${DOCKERFILE}" | grep -e "-LABEL version=\".*\"" | awk -F"\"" '{ print $2 }')
+  NEXT_VERSION=$(git diff origin/develop HEAD -- "${DOCKERFILE}" | grep -e "+LABEL version=\".*\"" | awk -F"\"" '{ print $2 }')
```

Comment:
I think this should be diffing `HEAD` with the base branch of the PR, not with `develop`. You can use `$GITHUB_BASE_REF` for that (thought I'd pass it to the script via a parameter instead of accessing it directly).

Reply:
I think we should diff against develop, that's the protected branch serving as ground truth for the images - we don't want PRs between arbitrary branches trigger any of this. Actually we should *additionally* restrict the triggers of the actions to only specifically PRs to develop as well.